### PR TITLE
120 memory leak

### PR DESCRIPTION
**Pull request by mpsijm**
This does not close the issue yet, but pull requests should be kept small and this enhancement reeaaalllyyy needs to be merged into master :)
I'm gonna refactor a bit more, in a separate branch, 120a. So far, the performance issues have been solved, only the ugly `instanceof` is still present :)

Have fun reviewing! :D

<performance,review

### CODE REVIEW

**Comment by MKerkhof** [comment]
Besides the @Override warning that IntelliJ throws which I already mentioned to you, everything looks oke :) The new smooth gameplay  is awesome and the code looks good! +1

**Comment by thervh70** [comment]
It looks really nice. The big overhaul of the render-engine seems good to me, it really has a positive effect on the gameplay.
This branch really has to be merged soon, however, as @MKerkhof said: IntelliJ doesn't like your @override. Please fix.

< review

File: src/main/java/nl/tudelft/ti2206/group9/util/ObservableLinkedList.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		/** Enumeration for the update types. */
+		enum Type {
+			/** When an element is added to the end of the list. */
+			ADD_LAST,
+			/** When an element is added to the begin of the list. */
+			ADD_FIRST,
+			/** When an element is removed from the list. */
+			REMOVE,
+			/** When an element is removed from the list by index. */
+			REMOVE_INDEX,
```
Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/util/ObservableLinkedList.java]: There is a comma at the end of this last enum. It isn't a big deal, but you could remove it ;)

**Comment by mdingjan** [comment]
Your changes are looking good! Playing the game is much more fun without lag ;)
Of course work is still in progress in terms of refactoring, but for now it is ready to be merged into master.

Please have a look at the small comments above and one of us will merge this branch.

<review

**Comment by mpsijm** [comment]
The @Overrides will be fixed in #128, so please leave that warning behind for now. I removed the comma as Mitchell requested, so in my opinion: ready to be merged! :)

<review,layout

**Pull request merged**